automation_testing/pages/SaveAsComponent.ts

```diff
@@ -0,0 +1,97 @@
+import { expect, Locator, Page } from '@playwright/test';
+import { url } from '../utils/formUrls';
+
+export class SaveAsComponent {
+    readonly page: Page;
+    readonly saveAsBtn: Locator;
+    readonly saveAsOptions: Locator;
+    readonly saveView: Locator;
+    readonly saveLabel: Locator;
+    readonly saveViewBtn: Locator;
+    readonly cancelViewBtn: Locator;
+    readonly viewBtn: Locator;
+    readonly saveAsCheckbox: Locator;
+    readonly tableViews: Locator;
+    readonly editView: Locator;
+    readonly editCheckbox: Locator;
+    readonly closeSaveViewBtn: Locator;
+    readonly closeEditViewBtn: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.saveAsBtn = page.locator('text=Save As');
+        this.saveAsOptions = page.locator('ul.MuiList-padding');
+        this.saveView = page.locator('//html/body/div[5]/div[3]/div/div');
+        this.saveLabel = page.locator('#label');
+        this.saveViewBtn = page.locator('div[role="presentation"] button:has-text("Save")');
+        this.cancelViewBtn = page.locator('div[role="presentation"] button:has-text("Cancel")');
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 8 spaces.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9498610)
```suggestion
		this.cancelViewBtn = page.locator('div[role="presentation"] button:has-text("Cancel")');
```